src/vuex/modules/data/store.js

```diff
@@ -37,20 +33,17 @@ export const mutations = {
   [EMPTY_NOTIFICATION] (state) {
     state.notifications = []
   },
-  [RECEIVE_COLLECTIONS] (state, collections) {
-    state.collections = collections
-  },
   [ADD_STORED_COLLECTION] (state, index, collection) {
     for (var i = 0; i < state.indexesAndCollections.length; i++) {
       if (state.indexesAndCollections[i].name === index) {
-        state.collections.stored.push(collection)
+        state.indexesAndCollections[i].collections.stored.push(collection)
```

Comment:
Yep, I'd really like to use array's functional API and avoid for loops, but these methods will always create a new array, which we'll have to use to replace the one in the store. Seems touchy considering Vue's reactivity mechanism (will need to use `Vue.set`).
Considering these factors, it seems overkill to do all this just to avoid a for loop...